### PR TITLE
Icds reports splited clean ui

### PR DESCRIPTION
**Pull request by dmydlo**
Hi @emord,
This is again PR with changes to UI code in ICDS Reports. This time broken into step-by-step commits for easier review.
I have closed the previous PR. (https://github.com/dimagi/commcare-hq/pull/20061)

### CODE REVIEW

**Comment by fyniac** [comment]
@czue can you please review?

File: custom/icds_reports/static/js/angular-services/baseControllers.service.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* global d3 */
+
+window.angular.module('icdsApp').factory('baseControllersService', function() {
```
Comment (czue) [inline custom/icds_reports/static/js/angular-services/baseControllers.service.js]: :+1: this seems like a great change

**Review by czue — APPROVED** [review]
this is a fantastic set of changes. :clap: 

the amount of code reuse you were able to take advantage of is mind-boggling :dizzy_face: and having it broken up by commit made it much much easier to make sense of, so thanks for doing that.

this front-end code has good test coverage right? I didn't review at the level where I would have caught small errors, but am thinking we can rely on tests for that, yeah? all the high-level restructures made good sense.

**Comment by dmydlo** [comment]
Yes, test coverage is quite good, each of the controllers has it's own test.

**Pull request merged**